services/alerta/service.go

```diff
@@ -100,16 +100,23 @@ func (s *Service) Update(newConfig []interface{}) error {
 }
 
 func (s *Service) Alert(token, resource, event, environment, severity, group, value, message, origin string, service []string, data interface{}) error {
+	c := s.config()
+
 	if resource == "" || event == "" {
 		return errors.New("Resource and Event are required to send an alert")
 	}
 
-	url, post, err := s.preparePost(token, resource, event, environment, severity, group, value, message, origin, service, data)
+	req, err := s.preparePost(token, resource, event, environment, severity, group, value, message, origin, service, data)
 	if err != nil {
 		return err
 	}
 
-	resp, err := http.Post(url, "application/json", post)
+	client := &http.Client{
```

Comment:
Can you create the client in the `NewService` method so that we don't have to create a new client for each new request?

Since the Alerta config can change dynamically you will also need to create a new client in the `Update` method if the config changed. And that also means that you will need to synchronize access to the client. For that I would recommend using atomic.Value, see how the `config` field is access for an example.

Let me know if I can clarify any of that better.